server/aesdsocket.c

```diff
@@ -0,0 +1,565 @@
+			
+			/********************************************************* 
+			*  STEP 3 : 
+			*  Receives data over the connection and 
+			*  appends to file /var/tmp/aesdsocketdata
+			*********************************************************/
+			
+			/*
+			*	-> receive a message from a socket
+			*	-> returns number of bytes received on success
+			*	   or -1 on error
+			*	-> sockfd 	- accepted socket file descriptor
+			*	-> buf[.len] 	- to store bytes received
+			*	-> len		- length of buffer
+			* 	-> flags	- MSG_WAITALL - Blocking mode
+			*/
+			do
+			{
+				recv_bytes = recv(accept_fd, buf, BUF_LEN, 0);
+				if(recv_bytes == RET_ERROR)
+				{
+					ERROR_LOG("Receive failed\n");
+					syslog(LOG_ERR,"Receive failed");
+					command_status.success = false;
+					break;
+				}
+				//buf[recv_bytes]='\0';
+				//DEBUG_LOG("Buffer : %ld",recv_bytes);
+				//DEBUG_LOG("Buffer : %s",buf);
+				
+				// write to file
+				ret = write(data_file_fd, buf, recv_bytes);
+				if(ret == RET_ERROR)
+				{
+					ERROR_LOG("File write failed\n");
+					syslog(LOG_ERR,"File write failed");
+					command_status.success = false;
+					break;
+				}
+			}while(buf[recv_bytes-1] != '\n');
+			if(command_status.success == false)
+			{
+				break;
+			}
+			
+			/********************************************************* 
+			*  STEP 4 : 
+			*  Returns the full content of /var/tmp/aesdsocketdata 
+			*  to the client as soon as the received data packet 
+			*  completes.
+			*  
+			*********************************************************/
+			
+			// lseek to start of the file
+			/*
+			*	-> send a message on a socket
+			*	-> returns number of bytes sent on success
+			*	   or -1 on error
+			*	-> fd 		- file descriptor
+			*	-> offset 	- file offset position
+			*	-> whence	- SEEK_SET, SEEK_CUR, SEEK_END
+			*/
+			off_t seek_ret = lseek(data_file_fd, 0, SEEK_SET);
+			if(seek_ret == RET_ERROR)
+			{
+				ERROR_LOG("lseek failed\n");
+				syslog(LOG_ERR,"lseek failed");
+				command_status.success = false;
+				break;
+			}
+			
+			// read from file
+			do
+			{
+				bytes_read = read(data_file_fd, read_buf, BUF_LEN);
+				if(bytes_read == RET_ERROR)
+				{
+					ERROR_LOG("File read failed\n");
+					syslog(LOG_ERR,"File read failed");
+					command_status.success = false;
+					break;
+				}
+				
+				//DEBUG_LOG("Bytes Read : %ld",bytes_read);			
+				//DEBUG_LOG("Read Buffer : \n%s",read_buf);
+				
+				/*
+				*	-> send a message on a socket
+				*	-> returns number of bytes sent on success
+				*	   or -1 on error
+				*	-> sockfd 	- sending socket file descriptor
+				*	-> buf[.len] 	- buffer with bytes to be sent
+				*	-> len		- length of buffer
+				* 	-> flags	- 0
+				*/
+				send_bytes = send(accept_fd, read_buf, bytes_read, 0);
+				if(send_bytes == RET_ERROR)
+				{
+					ERROR_LOG("Send failed\n");
+					syslog(LOG_ERR,"Send failed");
+					command_status.success = false;
+					break;
+				}
+			}while(read_buf[send_bytes-1] != '\n');
```

Comment:
@ameyflash I'm not sure this is the correct logic... I don't see handling for the case where the file doesn't end in `\n` and I think if you ever had a combination of packets whose total length was `BUF_LEN` this code would exit early before sending the remaining packets.